internal/blockstore/commit_and_query.go

```diff
@@ -268,11 +269,21 @@ func (s *Store) storeBlockHeaders(header *types.BlockHeader) error {
 		return errors.Wrapf(err, "can't calculate block hash {%d, %v}", number, header)
 	}
 
+	blockTx, err := block.GetTxIDs()
+	if err != nil {
+		return errors.Wrapf(err, "can't access block tx ids {%d, %v}", number, block)
+	}
+	txIdBytes, err := proto.Marshal(blockTx)
+	if err != nil {
+		return errors.Wrapf(err, "can't marshal block tx ids {%d, %v}", number, blockTx)
+	}
+
 	batch := &leveldb.Batch{}
 	batch.Put(constructHeaderBaseHashKey(number), blockHeaderBaseHash)
 	batch.Put(constructHeaderHashKey(number), blockHash)
 	batch.Put(constructHeaderBytesKey(number), blockHeaderBytes)
 	batch.Put(constructHeaderHashIndexKey(blockHash), encodeOrderPreservingVarUint64(number))
+	batch.Put(constructBlockTxKey(number), txIdBytes)
```

Comment:
Please rename all `BlockTx` to `BlockTxsID`. Initially, I got confused that we are storing the whole transaction. Let's be explicit.